src/nemos/basis.py

```diff
@@ -845,127 +886,136 @@ def evaluate_on_grid(self, n_samples: int) -> Tuple[NDArray, NDArray]:
         """
         return super().evaluate_on_grid(n_samples)
 
-
-class RaisedCosineBasisLinear(RaisedCosineBasis):
-    """Linearly-spaced raised cosine basis functions used by Pillow et al.
-
-    These are "cosine bumps" that uniformly tile the space.
-
-
-    Parameters
-    ----------
-    n_basis_funcs
-        Number of basis functions.
-
-    References
-    ----------
-    [^3]:
-        Pillow, J. W., Paninski, L., Uzzel, V. J., Simoncelli, E. P., & J.,
-        C. E. (2005). Prediction and decoding of retinal ganglion cell responses
-        with a probabilistic spiking model. Journal of Neuroscience, 25(47),
-        11003–11013. http://dx.doi.org/10.1523/jneurosci.3305-05.2005
-
-    """
-
-    def __init__(self, n_basis_funcs: int) -> None:
-        super().__init__(n_basis_funcs)
-
-    def _transform_samples(self, sample_pts: NDArray) -> NDArray:
-        """
-        Linearly map the samples from [0,1] to the the [0, (n_basis_funcs - 1) * pi].
-
-        Parameters
-        ----------
-        sample_pts :
-            The sample points used for evaluating the splines, shape (n_samples,)
-
-        Returns
-        -------
-        :
-            A transformed version of the sample points that matches the Raised Cosine basis domain,
-            shape (number of samples, ).
-        """
-        return sample_pts * np.pi * (self.n_basis_funcs - 1)
-
     def _check_n_basis_min(self) -> None:
         """Check that the user required enough basis elements.
 
-        Check that the number of basis is at least 1.
+        Check that the number of basis is at least 2.
```

Comment:
why did this change?

Reply:
the log-stretching needed at least 2 samples, in this way the two raised cosines behave coherently.